test/jobProcessor.js

```diff
@@ -150,3 +153,66 @@ describe('processUrlJob for a site', () => {
     });
   });
 });
+
+describe('broken sites', () => {
+  context('with a broken gzip', () => {
+    it('should print the error', () => {
+      let closePromise;
+
+      function getPort() {
+        return new Promise((resolve, reject) => {
+          portfinder.getPort((err, port) => {
+            if (err) {
+              return reject(err);
+            }
+            resolve(port);
+          });
+        });
+      }
+
+      function serveGzip(port, data) {
+        const options = {
+          key: fs.readFileSync('test/fixtures/server.key'),
+          cert: fs.readFileSync('test/fixtures/server.crt'),
+          requestCert: false,
+          rejectUnauthorized: false,
+        };
+
+        const server = https.createServer(options, (req, res) => {
+          res.writeHead(200, { 'Content-Encoding': 'gzip' });
+          res.end(data);
+          server.close();
+        }).listen(port);
+
+        closePromise = new Promise(resolve => {
+          server.on('close', resolve);
+        });
+
+        return new Promise(resolve => {
+          server.on('listening', resolve);
+        });
+      }
+
+      return new Promise((resolve, reject) => {
+        fs.readFile('./test/attachments/broken.gz', 'utf8', (err, data) => {
+          if (err) {
+            reject(err);
+            return;
+          }
+          resolve(data);
+        });
+      })
+      .then(data => getPort()
+        .then(port => {
+          process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
```

Comment:
I think you should revert this at some moment in the test. Maybe you could put this in a `before()` clause and restore in the `after()` clause.